.github/workflows/validate-graphql-codegen.yml

```
@@ -0,0 +1,49 @@
+name: Validate GraphQL CodeGeneration
+
+on:
+  pull_request:
+    branches:
+      - main
```

We can further restrict when this workflow runs by specifying the paths to watch. In this case, we want to monitor changes to the GraphQL schema file

```suggestion
on:
  pull_request:
    branches:
      - main
    paths:
      - 'schema.graphql'
 ```